lib/srv/forward/sshserver.go

```diff
@@ -726,79 +720,80 @@ func (s *Server) handleDirectTCPIPRequest(ch ssh.Channel, req *sshutils.DirectTC
 // handleSessionRequests handles out of band session requests once the session
 // channel has been created this function's loop handles all the "exec",
 // "subsystem" and "shell" requests.
-func (s *Server) handleSessionRequests(ch ssh.Channel, in <-chan *ssh.Request) {
+func (s *Server) handleSessionRequests(ctx context.Context, ch ssh.Channel, in <-chan *ssh.Request) {
 	// Create context for this channel. This context will be closed when the
 	// session request is complete.
 	// There is no need for the forwarding server to initiate disconnects,
 	// based on teleport business logic, because this logic is already
 	// done on the server's terminating side.
-	ctx, err := srv.NewServerContext(s.connectionContext, s, s.identityContext)
+
+	//nolint:ineffassign,staticcheck
```

Comment:
Since `NewServerContext` returns a `context.Context` as well as a `ServerContext`, I'm overwriting the value of the passed in context (`ctx`) variable to reflect the most up to date value (the newly created child context).

The linter rejects this since the `ctx` variable isn't current used after this point.  I could satisfy the linter by assigning the returned context variable to `_`, but I think this invites the risk of anyone who updates this function in the future accidentally using the old value of `ctx` since they may not realize they need to update this line to get the correct context value.

Reply:
Hmm, how about adding `ctx.Done()`  case to the select statement at the bottom of this method (in the for loop)?
And for `handleDirectTCPIPRequest` you could spawn a goroutine waiting on `ctx.Done()` to call `conn.Close()`